dns/pfSense-pkg-bind/files/usr/local/pkg/bind.inc

```diff
@@ -586,9 +586,24 @@ EOD;
 							$zone_conf .= "\n\n;\n;custom zone records\n;\n".base64_decode($zone['customzonerecords'])."\n";
 						}
 
+						// Freeze dynamic zones to prevent journal corruption
+						$zone_is_dynamic = $zone['enable_updatepolicy'] == "on" || $zoneallowupdate != "none";
```

Comment:
This could use some parenthesis around the logic to make it more clear what is going on. For example:
`(($zone['enable_updatepolicy'] == "on") || ($zoneallowupdate != "none"))`